### ISSUE TITLE
Definition of `const CACHE_LEVELS` causing precompilation to fail on Manjaro Linux

### ISSUE THREAD

**Issue by redpanda1234** [chriselrod/VectorizationBase.jl#32]
System information: 
```
Julia Version: 1.5.3 (2020-11-09)
OS: Manjaro Linux x86_64
Host: MacBookPro11,4 1.0
Kernel: 5.10.2-2-MANJARO
CPU: Intel i7-4980HQ (8) @ 4.000GHz
GPU: Intel Crystal Well
```
First noticed the issue when trying to precompile `DifferentialEquations.jl`.  Not sure if this is an issue with `VectorizationBase.jl` or with `Hwloc`...but the relevant line in `/src/topology.jl` is as follows: 
```julia
const CACHE_LEVELS = something(findfirst(isequal(0), CACHE_COUNT) - 1, length(CACHE_COUNT) + 1)
```
In particular, `findfirst(isequal(0), CACHE_COUNT)` throws an error if `0` does not occur in `CACHE_COUNT`. 

Here is the output on my machine after adding the lines 
```julia
@show CACHE_COUNT
@show findfirst(isequal(0), CACHE_COUNT)
```
to `/src/topology.jl`: 
```julia
julia> import VectorizationBase
[ Info: Precompiling VectorizationBase [3d5dd08c-fd9d-11e8-17fa-ed2836048c2f]
CACHE_COUNT = (4, 4, 1, 1)
findfirst(isequal(0), CACHE_COUNT) = nothing
ERROR: LoadError: LoadError: MethodError: no method matching -(::Nothing, ::Int64)
Closest candidates are:
  -(::BigInt, ::Union{Int16, Int32, Int64, Int8}) at gmp.jl:532
  -(::Base.CoreLogging.LogLevel, ::Integer) at logging.jl:117
  -(::Missing, ::Number) at missing.jl:115
  ...
Stacktrace:
 [1] top-level scope at /home/dipsticksupreme/.julia/dev/VectorizationBase/src/topology.jl:18
 [2] include(::Function, ::Module, ::String) at ./Base.jl:380
 [3] include at ./Base.jl:368 [inlined]
 [4] include(::String) at /home/dipsticksupreme/.julia/dev/VectorizationBase/src/VectorizationBase.jl:1
 [5] top-level scope at /home/dipsticksupreme/.julia/dev/VectorizationBase/src/VectorizationBase.jl:355
 [6] include(::Function, ::Module, ::String) at ./Base.jl:380
 [7] include(::Module, ::String) at ./Base.jl:368
 [8] top-level scope at none:2
 [9] eval at ./boot.jl:331 [inlined]
 [10] eval(::Expr) at ./client.jl:467
 [11] top-level scope at ./none:3
in expression starting at /home/dipsticksupreme/.julia/dev/VectorizationBase/src/topology.jl:18
in expression starting at /home/dipsticksupreme/.julia/dev/VectorizationBase/src/VectorizationBase.jl:352
ERROR: Failed to precompile VectorizationBase [3d5dd08c-fd9d-11e8-17fa-ed2836048c2f] to /home/dipsticksupreme/.julia/compiled/v1.5/VectorizationBase/Dto5m_zM0pN.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1305
 [3] _require(::Base.PkgId) at ./loading.jl:1030
 [4] require(::Base.PkgId) at ./loading.jl:928
 [5] require(::Module, ::Symbol) at ./loading.jl:923
```

Any quick fixes? Please lmk if there's any helpful information I could provide!

**Comment by chriselrod** [comment]
Oops. Should be a simple fix.

**Comment by chriselrod** [comment]
Should be fixed in VectorizationBase 0.14.12.
Try updating and let me know if there's a problem/everything works.

**Comment by redpanda1234** [comment]
Fixed!! Thanks :)
